report-topk-partion-sizes

```diff
@@ -351,13 +347,58 @@ def strfsize(num, suffix='B'):
         num /= 1024.0
     return "%.1f%s%s" % (num, 'Yi', suffix)
 
+def pretty_print(values):
+    """
+    Writes the report as a pretty-formatted list on stdout
+    """
+    header = ['Size', 'Partition', 'Size (in bytes)']
+    size = [len(x) for x in header]
+    results = []
+    # format the values and find out the largest element in each column
+    for val in values:
+        tup = (strfsize(val[1]), val[0], str(val[1]))
+        for idx, slen in enumerate(size):
+            if len(tup[idx]) > slen:
+                size[idx] = len(tup[idx])
+        results.append(tup)
+    results.insert(0, header)
+    # adjust the width of each cell
+    results = [[res[idx].ljust(s) for idx, s in enumerate(size)] for res in results]
+    # print it out
+    for res in results:
+        sys.stdout.write('%s  %s  %s\n' % (res[0], res[1], res[2]))
+
+def send_email(values, args, count, exec_start_time):
+    """
+    Create and send the report in an e-mail
+    """
+    print "Sending email report..."
+    # Craft an html email message report.
+    email = EmailMessage(
+        results=[(i[0], strfsize(i[1])) for i in values[:args.top]],
+        subject=subject(args.last_days),
+        email_address=args.email,
+        message_from="{}@{}".format(getpass.getuser(), local_hostname()),
+        cluster_name=args.cluster,
+        logstash_host=args.logstash_host,
+        execution_host=local_hostname(),
+        execution_time=time.time()-exec_start_time,
+        total_query_results=count,
+        unique_query_results=len(values)
+    )
+    email.attach(csv_attachement(values[:args.top]))    # Attach a csv file.
+    email.send()                                        # Deliver.
+    print "Done."
+
 def parse_arguments():
     """
     Parse command-line arguments.
     """
     parser = argparse.ArgumentParser(
         description="Generate an email report of the widest Cassandra partitions.")
-    parser.add_argument("-e", "--email", metavar="ADDRESS", required=True, help="Email address to send report to")
+    parser.add_argument("-e", "--email", metavar="ADDRESS", default='', help="Email the report to the specified address")
+    parser.add_argument('-s', '--csv', action='store_true', help='Output a comma-delimited report to stdout')
+    parser.add_argument('-t', '--printed', action='store_true', help='Output a pretty-formatted report to stdout')
```

Comment:
Not sure what you mean; My thinking was that printing a formatted report to stdout sounded like reasonable default behavior (i.e. in the absence of either email or csv).